test/integration/connect/envoy/Dockerfile-consul-envoy

```diff
@@ -0,0 +1,7 @@
+# Note this arg has to be before the first FROM
+ARG ENVOY_VERSION
+
+FROM consul-dev as consul
```

Comment:
The FROM consul-dev as consul is probably not needed. Instead in the makefile you can have a dependency on the linux target and just copy pkg/bin/linux_amd64/consul to /bin/consul.

That is how the dev-docker target works now. One issue with using `FROM consul-dev ...` is that it may or may not be the right image if you have specified CONSUL_DEV_IMAGE (which I do regularly for ent builds so as not to remove my oss image). You could make the consul image a build arg but really it would be just as easy to have a dep on the linux makefile target and copy in the binary from the host system.

Reply:
I think it's best like this - the way these integration tests work locally and in CI, the current `consul-dev` images _is_ the system under test.

Locally we assume you just built it (it's a dependency of the integration test make target) so it's the right one. In Circle CI we build it exactly as you said by using the linux binary from the workflow and copying it into the container.

We could in CI change it to do a copy in the same way as here but it would be effectively the same and would mean that locally you end up building twice. I guess make would prevent you actually rebuilding the linux target twice.

🤷‍♂ so ultimately I think it's pretty much equivalent either way but just a bit more poking/trial and error to make that work right in circle maybe? (the binary from a previous job is copied into the running container's dir but not in the same place that it would appear in a local build).

Reply:
It should be mostly equivalent except if you run `make ... CONSUL_DEV_IMAGE=consul-ent-dev` or with any non-default dev-image name to prevent stomping on whatever your existing consul-dev image is. Then all of this will pull the wrong thing and fail. Then again all the docker compose stuff using consul-dev so its just a limitation of the integration test tool.